packages/device_info_plus/device_info_plus_platform_interface/lib/model/android_device_info.dart

```diff
@@ -239,3 +247,63 @@ class AndroidBuildVersion {
     );
   }
 }
+
+
+/// Details for the current display
+///
+/// See: https://developer.android.com/reference/android/util/DisplayMetrics
+class AndroidDisplayMetrics {
+  const AndroidDisplayMetrics._({
+    required this.widthPx,
+    required this.heightPx,
+    required this.xDpi,
+    required this.yDpi,
+  });
+
+  /// Gets the absolute width in pixels of the largest region of the display accessible to an app
+  /// in the current system state, without subtracting any window decor or applying scaling factors.
+  final double widthPx;
+
+  /// Gets the absolute height in pixels of the largest region of the display accessible to an app
+  /// in the current system state, without subtracting any window decor or applying scaling factors.
+  final double heightPx;
+
+  /// The exact physical pixels per inch of the screen in the X dimension.
+  final double xDpi;
+
+  /// The exact physical pixels per inch of the screen in the Y dimension.
+  final double yDpi;
+
+  /// The exact physical display width in inches.
+  double get widthInches => widthPx / xDpi;
+
+  /// The exact physical display height in inches.
+  double get heightInches => heightPx / yDpi;
+
+  /// The exact physical size in inches measured diagonally across the display.
```

Comment:
It seems that we can't guarantee that the exact real size will be returned here and a different way should be used to calculate it: 
https://programmer.group/android-get-screen-physical-size-density-and-resolution.html

Reply:
Checked and it seems that current implementation is correct.